code/game/objects/structures/beds_chairs/bed.dm

```diff
@@ -52,29 +52,41 @@
 	icon_state = "down"
 	anchored = FALSE
 	resistance_flags = NONE
-	var/foldabletype = /obj/item/roller
+	/// Internal storage for the roller item that created us
+	var/obj/item/roller
 
-/obj/structure/bed/roller/attackby(obj/item/W, mob/user, params)
-	if(istype(W, /obj/item/roller/robo))
-		var/obj/item/roller/robo/R = W
-		if(R.loaded)
-			to_chat(user, span_warning("You already have a roller bed docked!"))
-			return
+/obj/structure/bed/roller/Initialize(mapload, obj/item/roller/roller)
+	. = ..()
+	if(!roller)
+		roller = new /obj/item/roller(src)
+	src.roller = roller
+	roller.forceMove(src)
+	name = roller.name
+
+/obj/structure/bed/roller/Destroy()
+	if(roller)
+		qdel(roller)
+		roller = null
+	return ..()
 
-		if(has_buckled_mobs())
-			if(buckled_mobs.len > 1)
-				unbuckle_all_mobs()
-				user.visible_message(span_notice("[user] unbuckles all creatures from [src]."))
-			else
-				user_unbuckle_mob(buckled_mobs[1],user)
-		else
-			R.loaded = src
-			forceMove(R)
-			user.visible_message(span_notice("[user] collects [src]."), span_notice("You collect [src]."))
-		return 1
-	else
+/obj/structure/bed/roller/attackby(obj/item/W, mob/user, params)
```

Comment:
use a more descriptive name than W please

Reply:
That's the base definition of the proc.